### PR TITLE
Fix bug in HorzIJ routine

### PR DESCRIPTION
**Pull request by bena-nasa**
Fix issue with HorzIJ routine. It had been generalized to work for the UFS/GOCART/NOAA work. However, I forgot to check the centers that are retrieved from the grid is those are degrees as well and convert those to radians too if needed.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [X] I have tested this change with a run of GEOSgcm (if non-trivial)
- [X] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [X] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Comment by rmontuoro** [comment]
While I could not test this branch since MAPL `develop` crashes UFS-GOCART (see issue #737), applying these changes to my MAPL fork ([bugfix/rmontuoro/ufs](https://github.com/rmontuoro/MAPL/tree/bugfix/rmontuoro%2Fufs) - cccff393) resulted in `SO2EMVN` still showing a pattern resembling the one in GOCART issue [#28](https://github.com/GEOS-ESM/GOCART/issues/28) during the first few output time steps. This pattern blends into a more realistic one as GOCART steps forward.

**Comment by tclune** [comment]
Hmm.  It is hard to see how there could be any transitory aspects of this layer.

**Comment by bena-nasa** [comment]
I found the bug, I was correcting the units on the arrays that get stuffed into the grid if those were not already attached but not the output arrays with the coordinates in the MAPL_GridGetCorners subroutine. On subsequent calls when we pull the corners from the grid, they are correct. This explains why the first time was wrong.

File: base/MAPL_Base.F90 [hunk 1]
```diff
@@ -2679,6 +2679,10 @@ subroutine MAPL_GridGetCorners(grid,gridCornerLons, gridCornerLats, RC)
 
          call ESMF_GridGet(grid,coordSys=coordSys,rc=status)
          _VERIFY(status)
+         if (coordSys==ESMF_COORDSYS_SPH_DEG) then
+            gridCornerLons=gridCornerLons*MAPL_DEGREES_TO_RADIANS
+            gridCornerLats=gridCornerLats*MAPL_DEGREES_TO_RADIANS
+         end if
```
Comment (tclune) [inline base/MAPL_Base.F90:2685]: Still needs an else clause for CARTESIAN case

File: base/MAPL_Base.F90 [hunk 2]
```diff
@@ -3310,7 +3310,8 @@ subroutine MAPL_GetHorzIJIndex(npts,II,JJ,lon,lat,lonR8,latR8,Grid,IMGlob,JMGlob
      real(kind=REAL64), parameter :: PI_R8     = 3.14159265358979323846_REAL64
```
Comment (tclune) [inline base/MAPL_Base.F90:3310]: Why can't we use MAPL_PI here?

File: base/MAPL_Base.F90 [hunk 3]
```diff
@@ -3347,14 +3348,27 @@ subroutine MAPL_GetHorzIJIndex(npts,II,JJ,lon,lat,lonR8,latR8,Grid,IMGlob,JMGlob
...
+        allocate(center_lons(im,jm),center_lats(im,jm))
+
+        if (coordSys==ESMF_COORDSYS_SPH_DEG) then
+           center_lons=lons*MAPL_DEGREES_TO_RADIANS
+           center_lats=lats*MAPL_DEGREES_TO_RADIANS
+        else if (coordSys==ESMF_COORDSYS_SPH_RAD) then
+           center_lons=lons
+           center_lats=lats
+        else if (coordSys==ESMF_COORDSYS_CART) then
+           _RETURN(_FAILURE)
```
Comment (tclune) [inline base/MAPL_Base.F90:3364]: Should be
```
_FAIL('Unsupported coordinate system:  ESMF_COORDSYS_CART')
```

**Review by tclune — CHANGES_REQUESTED** [review]
A few minor changes requested.

**Review by tclune — APPROVED** [review]

**Pull request merged**